paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -820,7 +820,7 @@ internal class PaymentSheetViewModel @Inject internal constructor(
     }
 
     override fun onUserCancel() {
-        reportDismiss()
+        eventReporter.onDismiss()
```

Comment:
Should we feed all analytics through through `AnalyticsListener`?

Reply:
I actually think it's better to have just the "logic" stuff in the listener, and make direct calls to the reporter in places that use it.

I had some things that contradicted that, but I updated it.

Reply:
One could argue that everything here should just move to `EventReporter` event.